tests/Validator/UniqueTypeNamesTest.php

```diff
@@ -0,0 +1,210 @@
+<?php
+
+declare(strict_types=1);
+
+namespace GraphQL\Tests\Validator;
+
+use GraphQL\Type\Schema;
+use GraphQL\Utils\BuildSchema;
+use GraphQL\Validator\Rules\UniqueTypeNames;
+
+class UniqueTypeNamesTest extends ValidatorTestCase
+{
+    /**
+     * @param array<int, array<string, mixed>> $errors
+     */
+    private function expectSDLErrors(string $sdlString, ?Schema $schema, array $errors): void
+    {
+        $this->expectSDLErrorsFromRule(new UniqueTypeNames(), $sdlString, $schema, $errors);
+    }
+
+    // Validate: Unique type names
```

Comment:
This should rather be attached to the first test case of the group, otherwise it can get lost during structural editing